core/src/subspace.rs

```diff
@@ -63,24 +63,23 @@ impl Subspace {
     pub(crate) fn split(&mut self) {
         self.state = State::Unavailable;
         let new_network_length = self.record.cidr.network_length() + 1;
-        let low_cidr: IpCidr;
-        let high_cidr: IpCidr;
-        match self.record.cidr {
+        let Some(cidrs) = (match self.record.cidr {
             IpCidr::V4(cidr) => {
-                let subnets = separator::Ipv4CidrSeparator::sub_networks(&cidr, new_network_length);
-                let subnets_vec = subnets.unwrap();
-                low_cidr = IpCidr::V4(*subnets_vec.first().unwrap());
-                high_cidr = IpCidr::V4(*subnets_vec.get(1).unwrap());
+                separator::Ipv4CidrSeparator::sub_networks(&cidr, new_network_length)
+                    .and_then(|vs| vs.as_slice().try_into().ok())
```

Comment:
Instead of using `ok()` could we do something that generates the error message here to address the coverage issue I mentioned? Like use `expect()`?